modules/hdf/samples/read_write_attributes.cpp

```diff
@@ -0,0 +1,105 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+/**
+ * @file read_write_attributes.cpp
+ * @author Fangjun Kuang <csukuangfj dot at gmail dot com>
+ * @date December 2017
+ *
+ * @brief It demonstrates how to read and write attributes inside the
+ * root group.
+ *
+ * Currently, only the following datatypes can be used as attributes:
+ *  - cv::String
+ *  - int
+ *  - double
+ *  - cv::InputArray (n-d continuous multichannel arrays)
+ *
+ * Although HDF supports associating attributes with both datasets and groups,
+ * only support for the root group is implemented by OpenCV at present.
+ */
+
+#ifdef __GNUC__
+#  pragma GCC diagnostic ignored "-Wmissing-declarations"
+#  if defined __clang__ || defined __APPLE__
+#    pragma GCC diagnostic ignored "-Wmissing-prototypes"
+#    pragma GCC diagnostic ignored "-Wextra"
```

Comment:
Add "static":
```
static void read_write_attributes()
```

Reply:
Thank you for the review. I have made changes according to you comments.

Reply:
Great! Please remove these "#pragma" lines too ("static" should resolve these warnings)

Reply:
Indeed! Warnings disappear.